twentyfourseven.game.php

```diff
@@ -235,6 +235,43 @@ protected function getAllDatas()
         // Playable spaces on the board
         $result['spaces'] = self::getPlayableSpaces();
 
+        // Tallies for both players stats
+        $result['tallies'] = self::getPlayersTally();
+
+        return $result;
+    }
+
+    /*
+        getPlayersTally:
+
+        Gather all the players stats.
+        Place the stats data into an array of arrays keyed first by player id and then stat.
+
+        The method is called:
+        - in the getAllDatas function
+        - in the newScores notif
+    */
+    protected function getPlayersTally()
+    {
+        $result = array();
+        $stats = array('turns_number', 'tally_sum_of_7', 'tally_sum_of_24', 'tally_run_of_3', 'tally_run_of_4', 'tally_run_of_5', 'tally_run_of_6', 'tally_set_of_3', 'tally_set_of_4', 'tally_bonus');

```

Comment:
I would turn this into private const STAT_KEYS = [ 'turns_number', 'tally_sum_of_7', ... ] at the beginning of the class and then reference it as self::STAT_KEYS in the loop. This is a minor style issue but allows the same constant array to be used in other parts of the class if you needed to iterate over the stats for some reason.